### DISCUSSION TITLE
iOS DataScannerViewController Implementation?

### DISCUSSION THREAD

**Discussion by LRufenacht** [dotnet/maui#17309] [General]
Not sure if this is the right place, but are there plans to add support for iOS' implementation of the `DataScannerViewController`?

I see that VNDocumentCameraViewController and it's delegate are supported and I got that to work ok.

We have customers that are asking for this functionality, so I was just curious what the plans are. It's been out for a few years now I believe.

https://developer.apple.com/documentation/visionkit/datascannerviewcontroller

**Comment by dadude999** [comment]
I noticed this was missing as well, so I went ahead and added one - find that [here](https://www.nuget.org/packages/DataScannerViewController.Ios.Maui/). I've only wrapped the most basic methods but I can confirm it does scan barcodes!
